### PR TITLE
[DEGA-1531] Improve calculation of time spent by boolean expression

### PR DESCRIPTION
**Pull request by arodionov53**
Improve calculation of time spent by boolean expression.  We need to measure time inside `nif` program. For this a new function was added which returns tuple of three elements - first two as in the original function and the third one - time spend during calculation.

### CODE REVIEW

File: c_src/betree.c [hunk 1]
```diff
@@ -859,6 +860,95 @@ static ERL_NIF_TERM nif_betree_search(ErlNifEnv* env, int argc, const ERL_NIF_TE
     return retval;
 }
```
Comment (mijkenator) [inline c_src/betree.c:862]: why you are not calling nif_betree_search inside nif_betree_search_t?

is checking input parameters and creating return tuple affect measurements a lot?

File: c_src/betree.c [hunk 2]
```diff
@@ -859,6 +860,95 @@ static ERL_NIF_TERM nif_betree_search(ErlNifEnv* env, int argc, const ERL_NIF_TE
...
+    ERL_NIF_TERM res = enif_make_list(env, 0);
+
+	for (size_t i = report->matched; i;) {
+		i--;
+		res = enif_make_list_cell(env, enif_make_uint64(env, report->subs[i]), res);
+	}
+
+    clock_gettime(clock_type, &done);
+    // Convert to microseconds
+    ErlNifSInt64 tspent = (done.tv_sec - start.tv_sec) * 1000000 + (done.tv_nsec - start.tv_nsec) / 1000;
```
Comment (vs-ads) [inline c_src/betree.c:881]: to calculate `tspent`, there should be a case to consider when `done.tv_nsec < start.tv_nsec`.
something like:
```
struct timespec diff;
long diff_nsec = done.tv_nsec - start.tv_nsec;
if (diff_nsec < 0) {
  diff.tv_sec = done.tv_sec - start.tv_sec - 1;
  diff.tv_nsec = 1000000000 + diff_nsec;
} else {
  diff.tv_sec = done.tv_sec - start.tv_sec;
  diff.tv_nsec = diff_nsec;
}
```
Then, convert `diff` to `tspent`.

Comment (arodionov53) [reply c_src/betree.c:881]: Thank you. Will change.

File: c_src/betree.c (hunk 1 above)
Comment (arodionov53) [reply c_src/betree.c:862]: I tried to bee as accurate as possible.  But I think your are right. Will change

File: c_src/betree.c (hunk 2 above)
Comment (vs-ads) [reply c_src/betree.c:881]: After discussion we came to the conclusion that the original calculation is correct and changes are not required.

**Review by mijkenator — APPROVED** [review]

**Review by vs-ads — APPROVED** [review]

**Pull request merged**